packages/outline-view/src/browser/outline-view-tree.ts

```diff
@@ -0,0 +1,47 @@
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject, postConstruct } from 'inversify';
+import { CompositeTreeNode, TreeModelImpl, TreeSelectionService, TreeExpansionService } from '@theia/core/lib/browser';
+
+@injectable()
+export class OutlineViewTreeModel extends TreeModelImpl {
+
+    @inject(TreeSelectionService) protected readonly selectionService: TreeSelectionService;
+    @inject(TreeExpansionService) protected readonly expansionService: TreeExpansionService;
+
+    @postConstruct()
+    protected init(): void {
```

Comment:
it does not look good, you should call `super.init`, otherwise it is very error prone if the super implementation changes, but this code is not updated.

What are you trying to achieve here?

Reply:
Here I am trying to override the event listener for `onExpansionDidChange`. In the super implementation, it would select the parent node, which should be avoided in the outline tree. Selecting a new node would cause the shift of focus on another symbol in monaco editor. 
https://github.com/theia-ide/theia/blob/af00d42b3d71b0d0b723cb970fefafb9c8500281/packages/core/src/browser/tree/tree-model.ts#L158-L162

Reply:
> In the super implementation, it would select the parent node

Please extract [this bit](https://github.com/theia-ide/theia/blob/af00d42b3d71b0d0b723cb970fefafb9c8500281/packages/core/src/browser/tree/tree-model.ts#L160-L164) to a template method with the default implementation and only override this method.

> which should be avoided in the outline tree

What should happen then? It seems that collapsed children nodes remained selected with this implementation. Is it expected?

Reply:
The listener code in super would select the node if it is collapsed && it is the parent of the currently selected node. It does not keep the selection on the node after collapse.